unified-installer/astra-unified-installer.sh

```diff
@@ -1117,12 +1179,177 @@ is_astra_registry() {
     local -r registry_url="$1"
     if [ -z "$registry_url" ]; then fatal "no registry_url given"; fi
 
-    if echo "$registry" | grep -q "cr." && echo "$registry" | grep -q "astra"; then
+    if echo "$registry_url" | grep -q "cr." && echo "$registry_url" | grep -q "astra"; then
         return 0
     fi
     return 1
 }
 
+# find_astra_registry loops through the given list of registries and returns the first one which matches the
+# pattern of an Astra registry.
+find_astra_registry() {
+    local -ar registries=("$@")
+    for reg in "${registries[@]}" ; do
+        if is_astra_registry "$reg"; then
+            echo "$reg"
+            return 0
+        fi
+    done
+    return 1
+}
+
+find_astra_registry_in_connector_images() {
+    find_astra_registry "$CONNECTOR_IMAGE_REGISTRY" "$NEPTUNE_IMAGE_REGISTRY" "$CONNECTOR_OPERATOR_IMAGE_REGISTRY"
+}
+
+find_astra_registry_in_trident_images() {
+    find_astra_registry "$TRIDENT_OPERATOR_IMAGE_REGISTRY" "$TRIDENT_AUTOSUPPORT_IMAGE_REGISTRY" \
+        "$TRIDENT_IMAGE_REGISTRY" "$TRIDENT_ACP_IMAGE_REGISTRY"
+}
+
+generate_kustomization_dirs_and_files() {
+    local -r kustomization_file="$1"
+    local -r add_as_resource_to_top_kustomization_file="${2:-"false"}"
+    local -r kustomization_dir="$(dirname "$kustomization_file")"
+
+    mkdir -p "$kustomization_dir"
+    if [ ! -f "$kustomization_file" ]; then
+        cat <<EOF > "$kustomization_file"
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+patches:
+transformers:
+generatorOptions:
+  disableNameSuffixHash: true
+secretGenerator:
+EOF
+        logdebug "$kustomization_file: OK"
+    fi
+
+    if [ "$add_as_resource_to_top_kustomization_file" == "true" ]; then
+        local -r top_kustomization_file="$__GENERATED_KUSTOMIZATION_FILE"
+        local -r base_dir="$(basename "$kustomization_dir")"
+        insert_into_file_after_pattern "$top_kustomization_file" "resources:" "- ./$base_dir"
+        logdebug "$top_kustomization_file: added resource entry for resource at path '$base_dir'"
+    fi
+}
+
+add_astra_registry_secret_to_kustomization() {
```

Comment:
This was also part of a step before; it was split out from the original step and made generic so that both the connector and trident components can generate the secret in their respective namespaces.